### DISCUSSION TITLE
can we make textual widgets thinner?

### DISCUSSION THREAD

**Discussion by braindevices** [Textualize/textual#4864] [Q&A] [answered]
the textual button, checkbox, etc always take multiple lines, this make the interface waste lots of space.
I want it to be more compact, I wonder is there any style control can get rid of the up an bottom margin and put them on a single line?

**Comment by kbernhagen** [comment]
https://textual.textualize.io/widgets/button/#additional-notes

You need to also style `:hover`

```python
from textual.app import App, ComposeResult
from textual.containers import Horizontal
from textual.widgets import Button

class Toolbar(Horizontal):
    DEFAULT_CSS = """
    Toolbar {
      height: 1;
      margin: 0;
      align: center middle;
    }
    Toolbar Button {
      margin: 0 1 0 0;
      padding: 0;
      height: 1;
      border: none;
      background: $primary;
      color: $text;
      text-style: none;
      &:hover {
          text-style: bold;
          padding: 0;
          border: none;
          background: $accent;
      }
    }
    """

class MyApp(App):
    def compose(self) -> ComposeResult:
        with Toolbar():
            yield Button("A")
            yield Button("B")
            yield Button("C")

if __name__ == "__main__":
    app = MyApp()
    app.run()
```

**Comment by willmcgugan** [comment, marked as answer]
> the textual button, checkbox, etc always take multiple lines, this make the interface waste lots of space. I want it to be more compact, I wonder is there any style control can get rid of the up an bottom margin and put them on a single line?

Most of the controls have a border, which you can remove with a few lines of CSS.

**Reply by braindevices** [reply]
thanks a lot yes the border is the culprit, I did following to remove all borders by default:
```
Static {
    border: none;
}
```
